### PR TITLE
istioctl proxy-config log: add label selector to filter logs

### PR DESCRIPTION
**Pull request by shamsher31**
To fix https://github.com/istio/istio/issues/27490

Based on @esnible  https://github.com/istio/istio/issues/27490#issuecomment-698413935

> istioctl does accept a --selector <selector> for some commands, e.g. istioctl dashboard envoy. There is no reason not to make some/all of the istioctl proxy-config commands take the same --selector argument if it will help.

Logs by directly passing pods name.
```
$ istioctl pc log istiod-5d884bcc7-mpbpt -n istio-system
2020-10-14T08:45:00.146435Z     error   klog    an error occurred forwarding 41023 -> 15000: error forwarding port 15000 to pod 8becc3bca0ef848c919139d7ecba423647ed738989d12abb5c17425ccd4a4948, uid : failed to execute portforward in network namespace "/var/run/netns/cni-f636d912-b685-538b-d549-29e6d71feb3d": failed to dial 15000: dial tcp4 127.0.0.1:15000: connect: connection refused
Error: failed to execute command on Envoy: failure running port forward process: Post "http://localhost:41023/logging": EOF
```

The same can be achieved by using `log -l app=istiod `.

```
$ istioctl pc log -l app=istiod -n istio-system
2020-10-14T08:45:04.621479Z     error   klog    an error occurred forwarding 33105 -> 15000: error forwarding port 15000 to pod 8becc3bca0ef848c919139d7ecba423647ed738989d12abb5c17425ccd4a4948, uid : failed to execute portforward in network namespace "/var/run/netns/cni-f636d912-b685-538b-d549-29e6d71feb3d": failed to dial 15000: dial tcp4 127.0.0.1:15000: connect: connection refused
Error: failed to execute command on Envoy: failure running port forward process: Post "http://localhost:33105/logging": EOF
```

### CODE REVIEW

**Comment by shamsher31** [comment]
@esnible Let me know if the `--selector` needs to be added for all `proxy-config` commands. I can implement that in a separate PR?

**Comment by shamsher31** [comment]
@esnible PTAL.

**Review by esnible — CHANGES_REQUESTED** [review]
When we set the log level, we probably should set it for ALL pods matching the selector, not just the first one we encounter that matches the selector.

File: releasenotes/notes/27490.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+issue:
+  - 27490
+
+releaseNotes:
+- |
+  **Added** `istioctl proxy-config log` now accept label selectors to filter logs based on label.
```
Comment (brian-avery) [inline releasenotes/notes/27490.yaml:9]:
```suggestion
  **Added** `istioctl proxy-config log` now accepts label selectors to filter logs based on label.
```

Comment (shamsher31) [reply releasenotes/notes/27490.yaml]: Done. Thanks

**Comment by shamsher31** [comment]
@esnible PTAL.

**Review by esnible — APPROVED** [review]
I think this is good.

It might be worthwhile fixing the output of `istioctl pc` to let the user know `pc log` can retrieve **or set** the log level.  This isn't the fault of your PR.

In the future, we may want want a follow-up so that `istioctl pc log deployment/httpbin` sets all of the log levels.  It will be confusing that all log levels can be set with `--selector`, but not by specifying a Deployment.

In the future, we should add an integration test for this to _tests/integration/pilot/istioctl_test.go_.

**Comment by shamsher31** [comment]
> It might be worthwhile fixing the output of istioctl pc to let the user know pc log can retrieve or set the log level.

@esnible Thanks, I will look into it in follow up PR.

**Comment by shamsher31** [comment]
@brian-avery PTAL.

**Review by brian-avery — APPROVED** [review]

**Comment by tariq1890** [comment]
Thanks for this!

**Pull request merged**